vcl/vcl_compiler_detection.h

```diff
@@ -1,314 +1,22 @@
 #ifndef VXL_COMPILER_DETECTION_H
 #define VXL_COMPILER_DETECTION_H
 
-#define VXL_DEC(X) (X)
-#define VXL_HEX(X) ( \
-    ((X)>>28 & 0xF) * 10000000 + \
-    ((X)>>24 & 0xF) *  1000000 + \
-    ((X)>>20 & 0xF) *   100000 + \
-    ((X)>>16 & 0xF) *    10000 + \
-    ((X)>>12 & 0xF) *     1000 + \
-    ((X)>>8  & 0xF) *      100 + \
-    ((X)>>4  & 0xF) *       10 + \
-    ((X)     & 0xF) \
-    )
+/*
+Remove deprecated compiler feature detection mechansims
 
-#ifdef __cplusplus
-# define VXL_COMPILER_IS_Comeau 0
-# define VXL_COMPILER_IS_Intel 0
-# define VXL_COMPILER_IS_IntelLLVM 0
-# define VXL_COMPILER_IS_PathScale 0
-# define VXL_COMPILER_IS_Embarcadero 0
-# define VXL_COMPILER_IS_Borland 0
-# define VXL_COMPILER_IS_Watcom 0
-# define VXL_COMPILER_IS_OpenWatcom 0
-# define VXL_COMPILER_IS_SunPro 0
-# define VXL_COMPILER_IS_HP 0
-# define VXL_COMPILER_IS_Compaq 0
-# define VXL_COMPILER_IS_zOS 0
-# define VXL_COMPILER_IS_XLClang 0
-# define VXL_COMPILER_IS_XL 0
-# define VXL_COMPILER_IS_VisualAge 0
-# define VXL_COMPILER_IS_NVHPC 0
-# define VXL_COMPILER_IS_PGI 0
-# define VXL_COMPILER_IS_Cray 0
-# define VXL_COMPILER_IS_TI 0
-# define VXL_COMPILER_IS_Fujitsu 0
-# define VXL_COMPILER_IS_GHS 0
-# define VXL_COMPILER_IS_SCO 0
-# define VXL_COMPILER_IS_ARMCC 0
-# define VXL_COMPILER_IS_AppleClang 0
-# define VXL_COMPILER_IS_ARMClang 0
-# define VXL_COMPILER_IS_Clang 0
-# define VXL_COMPILER_IS_GNU 0
-# define VXL_COMPILER_IS_MSVC 0
-# define VXL_COMPILER_IS_ADSP 0
-# define VXL_COMPILER_IS_IAR 0
-# define VXL_COMPILER_IS_MIPSpro 0
+Since cmake version 3.0 the use of The WriteCompilerDetectionHeader
+module is not recommended.
 
-#if defined(__COMO__)
-# undef VXL_COMPILER_IS_Comeau
-# define VXL_COMPILER_IS_Comeau 1
+Run "cmake --help-policy CMP0120" for policy details.
 
-#elif defined(__INTEL_COMPILER) || defined(__ICC)
-# undef VXL_COMPILER_IS_Intel
-# define VXL_COMPILER_IS_Intel 1
+Prevent configuration time warning messages when using
+cmake versions greater than 3.20
+*/
+#warning "The vcl_compiler_detection.h header file is no longer used.\n"   \
```

Comment:
This is likely to produce a compile error on MSVC. See https://stackoverflow.com/questions/29694305/preprocessor-warning-equivalent-in-visual-c.

Reply:
@dzenanz   I will put conditional code for MSVC around this before I merge.  Thank you.